cms/static/js/views/utils/create_library_utils.js

```diff
@@ -1,58 +1,17 @@
 /**
  * Provides utilities for validating libraries during creation.
  */
-define(["jquery", "underscore", "gettext", "js/views/utils/view_utils"],
-    function ($, _, gettext, ViewUtils) {
+define(["jquery", "gettext", "js/views/utils/view_utils", "js/views/utils/create_utils_base"],
+    function ($, gettext, ViewUtils, CreateUtilsFactory) {
         "use strict";
         return function (selectors, classes) {
-            var validateTotalKeyLength, setNewLibraryFieldInErr, hasInvalidRequiredFields,
-                createLibrary, validateFilledFields, configureHandlers;
+            var keyLengthViolationMessage = gettext("The combined length of the organization and library code fields cannot be more than <%=limit%> characters.")
+            var keyFieldSelectors = [selectors.org, selectors.number];
+            var nonEmptyCheckFieldSelectors = [selectors.name, selectors.org, selectors.number];
 
-            var validateRequiredField = ViewUtils.validateRequiredField;
-            var validateURLItemEncoding = ViewUtils.validateURLItemEncoding;
+            CreateUtilsFactory.call(this, selectors, classes, keyLengthViolationMessage, keyFieldSelectors, nonEmptyCheckFieldSelectors);
```

Comment:
This function is called via `new`. It's an object constructor. An object constructor need not return anything, its `this` becomes the return.

So:

    var thing = {'stuff': 1, 'more_stuff': 'hello'}

...is equivalent to:

    thing_maker = function () {
        this.stuff = 1;
        this.more_stuff = "hello";
    }
    var thing = new thing();

Instead of returning an object, we're returning a constructor, so this.create is part of the resulting object.